api/main.py

```diff
@@ -0,0 +1,50 @@
+from fastapi import FastAPI, HTTPException
+from fastapi.responses import JSONResponse
+from fastapi import BackgroundTasks
+import subprocess
+import os
+import json
+
+app = FastAPI()
+
+
+@app.get("/")
+def root():
+    return {"message": "Hello World"}
+
+
+@app.post("/chapters")
+async def generate_chapters(video_url: str, background_tasks: BackgroundTasks):
+    # Step 1: Run subtitles.py asynchronously
+    background_tasks.add_task(run_subtitles, video_url)
+
+    # Step 2: Return response
+    response_content = {
+        "message": "Generating chapters. Check back later for the result."
+    }
+    return JSONResponse(content=response_content, status_code=200)
+
+
+def run_subtitles(video_url: str):
+    # Step 1: Download video using youtube-dl or any other suitable tool
+    # Example: subprocess.run(["youtube-dl", video_url, "-o", "videos/video.mp4"])
+
+    # Step 2: Set environment variables
+    os.environ["INPUT_FILE_PATH"] = "./videos/video.mp4"
```

Comment:
Instead of setting the env variables here, wouldn't it be better to create a server.sh that can initialise envs and also make the uvicorn call to spin up the server at the end ?